test/browsers/wdio-mock.ts

```diff
@@ -0,0 +1,27 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import type { RemoteOptions } from 'webdriverio';
+
+class FakeWebdriver {
+  isMobile = true;
+  options: RemoteOptions;
+
+  constructor(options: RemoteOptions) {
+    this.options = options;
+  }
+
+  setTimeout() {
+    // noop
+  }
+}
+
+export const remoteMock = jest.fn();
+
+beforeEach(() => {
+  remoteMock.mockReset();
+  remoteMock.mockImplementation((options: RemoteOptions) => new FakeWebdriver(options));
+});
+
+jest.mock('webdriverio', () => ({
+  remote: remoteMock,
+}));
```

Comment:
Probably the biggest change in this PR

Before tests called private methods directly and asserted return value.

With typescript, the strategy changed and now tests use only public methods. Then we use this module mock to assert that webdriver received expected parameters and capabilities